allure-commandline/src/main/java/io/qameta/allure/Commands.java

```diff
@@ -70,6 +70,10 @@ public ExitCode generate(final Path reportDirectory,
         return ExitCode.NO_ERROR;
     }
 
+    private static boolean directoryIsNotEmpty(final Path reportDirectory) {
+        return Optional.ofNullable(reportDirectory.toFile().list()).map(l -> l.length).orElse(0) != 0;
```

Comment:
I don't like this logic, you can use `tmp.resolve("allure-report")` in serve command to make sure that directory doesn't exists